Account/FirefoxAccount.swift

```diff
@@ -190,13 +195,79 @@ open class FirefoxAccount {
             return "Not in a Token State: \(state?.label.rawValue ?? "Empty State")"
         }
     }
-
+    
+    public class FxAProfile {
+        open var avatarUrl: URL?
+        open var avatarImageData: Data?
+        open var displayName: String?
+        open let email: String
+        
+        init(email: String, displayName: String?, avatar: String?) {
+            self.email = email
+            self.displayName = displayName
+            self.avatarImageData = nil
+
+            if let avatarStr = avatar {
+                self.avatarUrl = URL(string: avatarStr)
+            } else {
+                self.avatarUrl = nil
+            }            
+        }
+        
+        func downloadAvatarImage() {
+            guard let avatarUrl = self.avatarUrl else {
+                return
+            }
+            
+            DispatchQueue.global().async {
+                self.avatarImageData = try? Data(contentsOf: avatarUrl)
```

Comment:
I'm not sure this specifies a `User-Agent`, or otherwise constructs requests as we expect. I know this is what our favicon fetches do, but it concerns me.